vortex-flatbuffers/flatbuffers/vortex-serde/footer.fbs

```diff
@@ -1,24 +1,18 @@
-table FlatLayout {
```

Comment:
do you just put begin/end for FlatLayout in the metadata?

Reply:
I'm not entirely sure this is desirable?

Having the format be aware of Messages (not necessarily flat array layouts) lets us do lots of nice things.

Perhaps the footer should have a big lookup table from message ID -> location?

I don't think the byte range should be hidden inside opaque metadata

Reply:
(An example of the nice things is if say one record batch needs messages 2 and 6, and we decide to coalesce the read, then we can know that we already have messages 3-5 if they're needed later.)

Reply:
The only difference imho between what was before and now is whether you need to read encoding id and construct the object out of it to read the range or can you embed that logic anywhere. We could decide to revert the flat layouts but in principle you can do this kind of optimization either way. The thing that could be useful though is that for chunked layouts we could not bother allocating the child flat layouts but we can do that now, just have to read the id

Reply:
No as in, I like this PR. I would just make messages the "layout" equivalent of array buffers.

So layouts have child layouts, _and_ child buffers/messages. The symmetry feels like it's appearing for a reason?